avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -806,7 +806,8 @@ private void processMap(JVar mapSchemaVar, final String name, final Schema mapSc
       );
 
       JBlock elseBlock = conditional._else();
-      elseBlock.assign(mapVar, codeModel.ref(Collections.class).staticInvoke("emptyMap"));
+      elseBlock.assign(mapVar, JExpr._new(schemaAssistant.classFromSchema(readerMapSchema, false))
```

Comment:
could this be made to generalize the keys for string props into CharSequence?

Reply:
This new logic is trying to reuse the same logic being used in fast-avro today to use specific charsequence type based on the schema.